test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smppy.py

```diff
@@ -0,0 +1,134 @@
+        estimator_parameters = {
+            "entry_point": smppy_mnist_script,
+            "role": "arn:aws:iam::920076894685:role/SageMakerRole",
+            "instance_count": 1,
+            "instance_type": INSTANCE_TYPE,
+            "framework_version": framework_version,
+            "hyperparameters": {"epochs": 1},
+            "profiler_config": ProfilerConfig(
+                smprofile_params=SMProfile(cpu_profiling_duration=3600)
+            ),
+            "debug_hook_config": False,
+        }
+        upload_s3_data_args = {"path": training_dir, "key_prefix": "pytorch/mnist"}
+        job_name = utils.unique_name_from_base("test-pt-smppy-training")
+        pytorch, _ = invoke_pytorch_estimator(
+            ecr_image,
+            sagemaker_regions,
+            estimator_parameters,
+            upload_s3_data_args=upload_s3_data_args,
+            job_name=job_name,
+        )
+        _check_and_cleanup_s3_output(pytorch, 40)
+
+
+@pytest.mark.usefixtures("feature_smppy_present")
+@pytest.mark.processor("gpu")
+@pytest.mark.integration("smppy")
+@pytest.mark.model("mnist")
+@pytest.mark.multinode(2)
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_training_smppy_distributed(framework_version, ecr_image, sagemaker_regions):
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        validate_or_skip_pytorchddp(ecr_image)
+        distribution = {"pytorchddp": {"enabled": True}}
+        estimator_parameters = {
+            "entry_point": smppy_mnist_script,
+            "role": "arn:aws:iam::920076894685:role/SageMakerRole",
+            "instance_count": 2,
+            "instance_type": INSTANCE_TYPE,
+            "framework_version": framework_version,
+            "distribution": distribution,
+            "hyperparameters": {"epochs": 1},
+            "profiler_config": ProfilerConfig(
+                smprofile_params=SMProfile(cpu_profiling_duration=3600)
+            ),
+            "debug_hook_config": False,
+        }
+        upload_s3_data_args = {"path": training_dir, "key_prefix": "pytorch/mnist"}
+        job_name = utils.unique_name_from_base("test-pt-smppy-training")
+        pytorch, _ = invoke_pytorch_estimator(
+            ecr_image,
+            sagemaker_regions,
+            estimator_parameters,
+            upload_s3_data_args=upload_s3_data_args,
+            job_name=job_name,
+        )
+        _check_and_cleanup_s3_output(pytorch, 60)
+
+
+def _check_and_cleanup_s3_output(estimator, wait_time):
+    s3 = boto3.client("s3")
+    bucket = estimator.output_path.replace("s3://", "").rstrip("/")
+
+    # Give postprocessing rule some time to complete
+    time.sleep(wait_time)
+
+    postproc_contents = s3.list_objects_v2(
+        Bucket=bucket, Prefix=_get_detailed_profiler_output_prefix(estimator)
+    ).get("Contents")
+    assert len(postproc_contents) > 0
+    for file in postproc_contents:
+        assert file.get("Size") > 0
+
+    return
+    all_contents = s3.list_objects_v2(
```

Comment:
This code will located after return. Do we need to cleanup resources?

Reply:
the return statement is a relic of my local testing since i still wanted to inspect output. i asked asimov oncall and he did suggest we should clean up resources so i added the code that cleans up the training job output.